packages/synthetics-sdk-broken-links/src/broken_links_func.ts

```diff
@@ -172,6 +113,37 @@ export async function retrieveLinksFromPage(
   );
 }
 
+/**
+ * Checks a list of links on a Puppeteer page, records the results, and throws a
+ * detailed if necessary.
+ *
+ * @param page - The Puppeteer page on which to check the links.
+ * @param links - An array of links to check.
+ * @param options - global options object with all broken link checker options.
+ * @returns A Promise that resolves with an array of successfully checked link
+ *          results
+ * @throws {Error} If an error occurs while checking the links, it throws an
+ *          error with an appropriate message, including which link errored.
+ */
+export async function checkLinks(
+  page: Page,
+  links: LinkIntermediate[],
+  options: BrokenLinksResultV1_BrokenLinkCheckerOptions
+): Promise<BrokenLinksResultV1_SyntheticLinkResult[]> {
+  const followed_links: BrokenLinksResultV1_SyntheticLinkResult[] = [];
+  for (const link of links) {
+    try {
+      followed_links.push(await checkLink(page, link, options));
+    } catch (err) {
+      if (err instanceof Error) process.stderr.write(err.message);
```

Comment:
Any error thrown by the actual link navigation is caught by in `fetchLink`, this extra try-catch is in order to surround all asynchronous calls with a try-catch in case something totally unexpected happens.

Meaning if an error was ever thrown by checkLink that isn't from the link navigation we don't know how to handle it.